CMakeLists.txt

```diff
@@ -84,8 +92,21 @@ macro(build_libyaml)
   set(yaml_LIBRARIES yaml)
 endmacro()
 
-build_libyaml()
-set(extra_test_dependencies libyaml-0.2.5)
+# Skip building yaml if the expected version is already present in the system
+if(yaml_FOUND)
+  if("${yaml_VERSION}" VERSION_EQUAL 0.2.5)
+    set(_SKIP_YAML_BUILD 1)
+  else()
+    message(WARNING
```

Comment:
I'm not sure we want this to be a warning. An info-level message is probably sufficient.

Reply:
My first idea was for this to be FATAL... the reason being that if you have libyaml in your system it means that you are already using it and having two versions in your system is the recipe for a disaster.

This obviously is geared towards a real robotics application use-case.
If you run ROS 2 on your laptop then you will definitely have packages installed that you don't use.